src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonObject.IDictionary.cs

```diff
@@ -74,7 +79,15 @@ public void Clear()
         /// <exception cref="ArgumentNullException">
         ///   <paramref name="propertyName"/> is <see langword="null"/>.
         /// </exception>
-        public bool ContainsKey(string propertyName) => Dictionary.ContainsKey(propertyName);
+        public bool ContainsKey(string propertyName)
+        {
+            if (propertyName is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(propertyName));
```

Comment:
This check is here to have a resulting exception contain "propertyName" instead of "key" in the exception instance?

Reply:
Correct. I shoehorned a fix for #103715 in this PR as a separate commit.